docs/rfc/rfc-005-accounts-auth.md

```diff
@@ -0,0 +1,301 @@
+sender actors cannot impersonate others.
+
+This leads to significant implications: Externally Owned Accounts (EOA) and ModuleAccounts (system-owned identities) can
+be treated as the same entity. Authorization, which determines which actor can execute a particular state transition
+towards another actor, must be managed by the recipient of the state transition. For example, the 'bank' module decides
+if the 'mint' module is permitted to inflate a coin's supply. This eliminates the need for permissions in the 'auth' 
+module as they currently exist.
+
+The key difference between EOA and ModuleAccount lies in the origin of state transitions on their behalf. In the case of
+ModuleAccount, a state transition cannot be initiated from a transaction (TX), as no valid credential can prove the sender
+is the ModuleAccount itself.
+
+We can distinguish between EOA and ModuleAccount by considering an account as a ModuleAccount (or SystemOwnedAccount)
+when it is not possible to initiate a state transition on its behalf from a TX. Given that this distinction occurs during
+the VerifySignature stage of the AnteHandler and that authorization is managed by the state transition recipient, the
+execution runtime or other system actors to manage differences between account types. Instead, they should be treated as
+business logic units capable of sending and receiving messages. This further proves that there's no need to distinguish
+`ModuleAccounts` from `EOA`.
+
+### Account Definition
+
+We define the new `Account` type, which is what an account needs to implement to be treated as such.
+An `Account` type is defined at APP level, so it cannot be dynamically loaded as the chain is running without upgrading the 
+node code, unless we use something like `x/wasm`. 
+
+```go
+type InternalAccount struct {
+	init    func(ctx *Context, msg proto.Message) (*InitResponse, error)
+	execute func(ctx *Context, msg proto.Message) (*ExecuteResponse, error)
+	query   func(ctx *Context, msg proto.Message) (proto.Message, error)
+	schema  func() *Schema
+}
+```
+
+This is an internal view of the account as intended by the system. It is not meant to be what developers implement. An
+example implementation of the `InternalAccount` type can be found in [this](https://github.com/testinginprod/accounts-poc/blob/main/examples/recover/recover.go) example of account whose credentials can be
+recovered. In fact, even if the `Internal` implementation is untyped (with respect to `proto.Message`), the concrete implementation is fully typed.
+
+During any of the execution methods of `InternalAccount`, `schema` excluded, the account is given a `Context` which provides:
+- A namespaced `KVStore` for the account, which isolates the account state from others (NOTE: no `store keys` needed, the account address serves as `store key`).
+- Information regarding itself (its address)
+- Information regarding the sender.
+- ...
+
+#### Init
```

Comment:
on the last call we talked about how an account will not be automatically instantiated, can we touch on this and the implications? 

If a user sends funds from an exchange to the wallet, the exchange would need to support creating the account.